packages/docs-app/src/examples/core-examples/treeExample.tsx

```diff
@@ -14,82 +14,116 @@
  * limitations under the License.
  */
 
-import React from "react";
+import React, { useCallback, useReducer } from "react";
 
 import { Classes, Icon, Intent, TreeNodeInfo, Tree } from "@blueprintjs/core";
 import { Example, ExampleProps } from "@blueprintjs/docs-theme";
-import { Tooltip2 } from "@blueprintjs/popover2";
+import { Classes as Popover2Classes, ContextMenu2, Tooltip2 } from "@blueprintjs/popover2";
 
-export interface TreeExampleState {
-    nodes: TreeNodeInfo[];
+type NodePath = number[];
+
+type TreeAction =
+    | { type: "SET_IS_EXPANDED"; payload: { path: NodePath; isExpanded: boolean } }
+    | { type: "DESELECT_ALL" }
+    | { type: "SET_IS_SELECTED"; payload: { path: NodePath; isSelected: boolean } };
+
+function forEachNode(nodes: TreeNodeInfo[], callback: (node: TreeNodeInfo) => void) {
+    if (nodes == null) {
```

Comment:
`nodes` are `TreeNodeInfo[]` so they're never null (unless `strictNullChecks` is off?)

Reply:
hm I think you're right, and it's a copy-pasta remnant of old code from when `strictNullChecks` was off

Reply:
actually, strictNullChecks is off for docs-app. will track fixing that separately